.github/workflows/pipelines-unlock.yml

```diff
@@ -0,0 +1,82 @@
+name: Pipelines
+run-name: Unlock Gruntwork Pipelines
+on:
+  workflow_call:
+    inputs:
+      lock_id:
+        description: "The ID of the lock, usually a GUID"
+        required: true
+        type: string
+      working_directory:
+        description: "Path to the terragrunt.hcl file where the lock is held"
+        required: true
+        type: string
+
+      # This field can be overriden to customize the runner used for pipelines
+      # workflows.
+      #
+      # IMPORTANT: To use self-hosted runners this workflow must be hosted in
+      # the same GitHub organization as your infra-live repository.
+      # See https://docs.github.com/en/actions/using-workflows/reusing-workflows#using-self-hosted-runners
+      #
+      # The value must be an escaped JSON string that will be decoded to the
+      # jobs.runs-on field
+      # See https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idruns-on
+      #
+      # For example:
+      # - A simple github runner: "\"ubuntu-22.04\""
+      # - A list of labels: "[\"self-hosted\", \"linux\"]"
+      # - A map: "{group: \"ubuntu-runners\", labels: \"ubuntu-20.04-16core\"}"
+      runner:
+        type: string
+        default: '"ubuntu-latest"'
+    secrets:
+        PIPELINES_READ_TOKEN:
+          required: true
+env:
+  PIPELINES_CLI_VERSION: v0.20.0
+  PIPELINES_ACTIONS_VERSION: event
+
+jobs:
+  unlock:
+    name: Unlock TF State
+    runs-on: ${{ fromJSON(inputs.runner) }}
+    steps:
+      - name: Checkout Pipelines Actions
+        uses: actions/checkout@v4
+        with:
+          path: pipelines-actions
+          repository: gruntwork-io/pipelines-actions
+          ref: ${{ env.PIPELINES_ACTIONS_VERSION }}
+          token: ${{ secrets.PIPELINES_READ_TOKEN }}
+
+      - name: Check out repo code
+        uses: actions/checkout@v4
+        with:
+          path: infra-live-repo
+          fetch-depth: 0
+
+      - name: Bootstrap Workflow
+        id: gruntwork_context
+        uses: ./pipelines-actions/.github/actions/pipelines-bootstrap
+        with:
+          token: ${{ secrets.PIPELINES_READ_TOKEN }}
+          working_directory: ${{ inputs.working_directory }}
+          terragrunt_command: "force-unlock -force ${{ inputs.lock_id }}"
+          branch: 'main'
+
+      - name: "Run terragrunt force-unlock in ${{ inputs.working_directory }}"
```

Comment:
It is odd to me that we're doing all this instead of just running:
```yml
  - name: "Force unlock"
    env:
      LOCK_ID: ${{ inputs.lock_id }}
      TERRAGRUNT_AUTH_PROVIDER_CMD: "pipelines auth terragrunt-credentials --ci github-actions --cloud aws --wd ."
    run: terragrunt force-unlock -force "$LOCK_ID"
```

What does using `pipelines-execute` buy us here?

Reply:
1) It seemed the idomatic way to call terragrunt in pipelines
2) I was concerned about whether or not we would get the AUTH_PROVIDER_CMD magic to work correctly without a regular pipelines execute run, however I did not test it to confirm.

Reply:
Ah, it gets us mise.  We don't actually have TG installed yet.